types/src/lib.rs

```diff
@@ -68,3 +68,26 @@ pub struct SendTransactionRequest {
 pub enum SendTransactionError {
     MalformedTransaction,
 }
+
+#[derive(CandidType, Debug, Deserialize, PartialEq)]
+pub enum UpdateTransactionError {
+    UnmodifiableTransaction,
+    UnknownTxId,
+    OneTransactionUpdateAlreadyConfirmed,
```

Comment:
Because as we discussed we just work with UTXOs we can't know precisely which transaction got confirmed so by using `TransactionAlreadyConfirmed` the user may think that the transaction identified by the provided he is trying to update got confirmed but in reality it may even be a previous one.

For instance:

let txid = send(amount, fees, recipient, true).await;

let result = update_transaction(amount, fees + 10_000, recipient).await;
let update_txid;
match result {
    Ok(txid) => update_txid = txid,
    Err(_) => panic!("update transaction error")
}

let result = update_transaction(update_txid, amount, fees + 20_000, recipient).await;
match result {
    Ok(txid) => (),
    Err(TransactionAlreadyConfirmed) => panic!("transaction already confirmed") // the user may think that the transaction identified by update_txid got confirmed but the library doesn't know. The library only knows if one of transactions identified by txid and updated_txid are confirmed not which one
    Err(_) => panic!("update transaction error")
}

Reply:
Can we know which transaction was confirmed?
In that case, it should be `TransactionAlreadyConfirmed(tx_id)`, indicating which transaction got confirmed.

Reply:
Unfortunately if we just restrict ourselves to `get_utxos` without any argument it seems impossible.

As I sent you on Slack:

In a scenario like the following one where a canister send some BTCs to a BTC address and later on want to know which transaction got confirmed, it's not possible to guess which from the original or updated transaction got confirmed.

> get_utxos()
<{
	{
		outpoint: {
			txid: txid,
			vout: vout0
		},
		value: k,
		height: m,
		confirmations: n
	},
	{
		outpoint: {
			txid: txid,
			vout: vout1
		},
		value: k,
		height: m,
		confirmations: n
	},
	{
		outpoint: {
			txid: txid,
			vout: vout2
		},
		value: k,
		height: m,
		confirmations: n
	}
}
> txid = send(2k, k, recipient);
> updated_txid = update_transaction(txid, k, 2k, recipient);
> get_utxos()
< {}

Reply:
As we discussed, we can ignore this issue for now.
It might be possible later on to use a transaction info API to gather information about the transactions.